### ISSUE TITLE
Changing `ospsuite`'s `linesColor` palette modifies identity and foldDistance lines colors.

### ISSUE THREAD

**Issue by Felixmil** [Open-Systems-Pharmacology/TLF-Library#472]
In ospsuite, when the `linesColor` palette is changed, the `foldDistance` and `identity` lines also change:
``` r
devtools::load_all()
#> ℹ Loading ospsuite
#> Loading required package: rClr
#> 
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000

simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
sim <- loadSimulation(simFilePath)

simResults <- runSimulation(sim)

obsData <- lapply(
  c("ObsDataAciclovir_1.pkml", "ObsDataAciclovir_2.pkml", "ObsDataAciclovir_3.pkml"),
  function(x) loadDataSetFromPKML(system.file("extdata", x, package = "ospsuite"))
)

names(obsData) <- lapply(obsData, function(x) x$name)

outputPaths <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
myDataCombined <- DataCombined$new()

# Add simulated results
myDataCombined$addSimulationResults(
  simulationResults = simResults,
  quantitiesOrPaths = outputPaths,
  groups = "Aciclovir PVB"
)

# Add observed data set
myDataCombined$addDataSets(obsData$`Vergin 1995.Iv`, groups = "Aciclovir PVB")

DPC <- DefaultPlotConfiguration$new()

DPC$linesColor = "red"
# DPC$pointsColor = "green"


plotObservedVsSimulated(myDataCombined, foldDistance = 2, defaultPlotConfiguration = DPC)
```

![](https://i.imgur.com/DBThWbP.png)

<sup>Created on 2023-06-29 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by Felixmil** [comment]
Based on my analysis of {tlf}, I've observed that the `plotObject$plotConfiguration$lines` setting, to where the `ospsuite` `linesColor `parameter go, is primarily used for two specific situations:

1. Drawing diagonal (identity, foldlines), horizontal, and vertical lines (LLOQ, y=0 lines).
2. Drawing the smoother lines. It's worth noting that the smoother lines exist in `{tlf}`, but they seem to not be used in ospsuite.
In the first case, these lines are not associated with any variables in the data. They are considered "static" and should not derive their colors from data groups.

In the second case, the smoother lines are applied to each group in the data, and therefore, they should obtain colors from the respective groups (similar to how scatter points are handled).

To summarize, modifying the `linesColor `parameter in `ospsuite `will currently only impact the color of the "static" lines.

However, it's important to consider that in the future, the `plotObject$plotConfiguration$lines` setting might be utilized for other purposes.  In such cases, should we create a separate setting specifically for what I refer to as "static lines."

What is your opinion on this @pchelle ?

**Comment by pchelle** [comment]
It may indeed be more transparent to expose the properties as actual named field (such as smoother and foldLines)
so that users could more easily explore and tune the property.

This may require some fixes down the road because of the integration of `{tlf}` with `{ospsuite}` and `{ospsuite.reportingengine}` packages

@Yuri05 @PavelBal thoughts ?

**Comment by PavelBal** [comment]
@pchelle From the user's perspective, I want to change the default color palette used to present the data sets. What I do (simplified) is:

```
DPC <- DefaultPlotConfiguration$new()

newColors <- c("red", "green", "blue")

DPC$linesColor = newColors
DPC$pointsColor = newColors
```

The side effect is that in the simulated-vs-observed, residuals-vs-time, and residuals-vs-simulated, the identity and fold-lines are not black any more, which is not the desired effect. Actually I do not see a use case where the user would like to change the color of those lines.

**Comment by pchelle** [comment]
@PavelBal 
In the simplified code, you actually set the colors of the lines to $\color{red}{red}$, $\color{green}{green}$ and $\color{blue}{blue}$ using **`DPC$linesColor = newColor`** or is this supposed to change the color of the errorbars ?
In which case, the `tlf::plotConfiguration` field is **`errorbars`** (but I don't know how it is integrated to `{ospsuite}`

**Comment by PavelBal** [comment]
> In the simplified code, you actually set the colors of the lines to 
, 
 and 
 using DPC$linesColor = newColor or is this supposed to change the color of the errorbars ?

I simply want to change the overall appearance, so when I create an individual time profile (or a population, whatever), the default colors are the ones I provide. What do I have to set?

**Comment by pchelle** [comment]
Is it because `ospsuite::DefaultPlotConfiguration` sets the properties for all the plots including time profile and obs vs pred plots ?

Then, the best fix would be in my opinion, to prevent `ospsuite::DefaultPlotConfiguration$linesColor` to update the property `tlf::ObsVsPredPlotConfiguration$lines$color` and keep the property defined by the current `tlf` theme.

**Comment by pchelle** [comment]
The new colors would be defined in both `ospsuite::DefaultPlotConfiguration$pointsColor` and `ospsuite::DefaultPlotConfiguration$linesColor`, wouldn't they ?

So, if `tlf::ObsVsPredPlotConfiguration$points$color` get the property from `ospsuite::DefaultPlotConfiguration$pointsColor`, the scatter plot will still follow the appropriate behavior while the identity and fold lines will just use the default that wrote you did not want to change.

**Comment by PavelBal** [comment]
Ok got it now. Thanks for clarification.

**Comment by pchelle** [comment]
> Ok got it now. Thanks for clarification.

No problem :) 
Since the naming is similar I think it is really hard to split what property is `{tlf}`, what property is `{ospsuite}` and how they are currently integrated to each other.

**Comment by Felixmil** [comment]
Since `ospsuite::DefaultPlotConfiguration$linesColor` only impact static lines which we don't want to change, why not forget about it and use only `ospsuite::DefaultPlotConfiguration$pointsColor` to set a custom "group palette" ? Smooth, error bars etc will follow this palette just like the points.

example:

``` r
devtools::load_all()
#> ℹ Loading ospsuite
#> Loading required package: rClr
#> 
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000
#> Warning: package 'testthat' was built under R version 4.3.1

# load the simulation
sim <- loadTestSimulation("MinimalModel")
simResults <- importResultsFromCSV(
  simulation = sim,
  filePaths = getTestDataFilePath("Stevens_2012_placebo_indiv_results.csv")
)

# import observed data (will return a list of `DataSet` objects)
dataSet <- loadDataSetsFromExcel(
  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012.xlsx"),
  importerConfiguration = loadDataImporterConfiguration(getTestDataFilePath("ImporterConfiguration.xml"))
)

# create a new instance and add datasets
myCombDat <- DataCombined$new()
myCombDat$addDataSets(dataSet)
myCombDat$addSimulationResults(
  simResults,
  quantitiesOrPaths = c(
    "Organism|Lumen|Stomach|Metformin|Gastric retention",
    "Organism|Lumen|Stomach|Metformin|Gastric retention distal",
    "Organism|Lumen|Stomach|Metformin|Gastric retention proximal"
  )
)

myCombDat$setGroups(
  names = c(
    "Organism|Lumen|Stomach|Metformin|Gastric retention",
    "Organism|Lumen|Stomach|Metformin|Gastric retention distal",
    "Organism|Lumen|Stomach|Metformin|Gastric retention proximal",
    "Stevens_2012_placebo.Placebo_total",
    "Stevens_2012_placebo.Placebo_distal",
    "Stevens_2012_placebo.Placebo_proximal"
  ),
  groups = c("Solid total", "Solid distal", "Solid proximal", "Solid total", "Solid distal", "Solid proximal")
)

# Plot Configuration
myPlotConfiguration <- DefaultPlotConfiguration$new()

# Change default color here
myPlotConfiguration$pointsColor <- viridis::viridis(3)

# Plot output
plotObservedVsSimulated(myCombDat, myPlotConfiguration, foldDistance = c(2,3,4))

```

![](https://i.imgur.com/Dfmqjd2.png)

<sup>Created on 2023-07-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by Felixmil** [comment]
@Yuri05 as deuteranopian, I can relate 😅
